app/models/users_role.rb

```diff
@@ -24,4 +30,14 @@ def self.current_role_for(user)
 
     nil
   end
+
+  # @param user [User]
+  # @param role [Role]
+  # @return [Boolean]
+  def self.set_last_role_for(user, role)
+    users_role = UsersRole.find_by(user: user, role: role)
+    return false if users_role.nil?
+
+    user.update(last_role_id: users_role.id)
```

Comment:
A bit confused why we're setting the last\_role\_id as the `users_role` rather than just the `role` itself?

Reply:
This way if a user has a particular last role and then that role is revoked, it gets removed in both places. From `user.roles` and `user.last_role`.

 See https://github.com/rubyforgood/human-essentials/blob/1bd860578d8c36fdad21d49621e0b9d2b3c93ced/spec/models/users_role_spec.rb#L64